lib/src/widgets/three_d_s_webview.dart

```diff
@@ -21,28 +52,44 @@ class ThreeDSWebView extends StatelessWidget {
         child: SizedBox(
           height: MediaQuery.of(context).size.height,
           width: MediaQuery.of(context).size.width,
-          child: WebView(
-            initialUrl: transactionUrl,
-            javascriptMode: JavascriptMode.unrestricted,
-            onPageFinished: (pageUrl) {
-              final redirectedTo = Uri.parse(pageUrl);
-              final callbackUri = Uri.parse(callbackUrl);
-
-              final bool hasReachedFinalRedirection =
-                  redirectedTo.host == callbackUri.host;
-
-              if (hasReachedFinalRedirection) {
-                final queryParams = redirectedTo.queryParameters;
-
-                String? status = queryParams['status'];
-                String? message = queryParams['message'];
-
-                on3dsDone(status, message);
-              }
-            },
+          child: WebViewWidget(
+            controller: _controller,
           ),
         ),
       ),
     );
   }
+
+  void _onProgress(int progress) {
+    debugPrint('WebView is loading (progress : $progress%)');
+  }
+
+  void _onPageStarted(String url) {
+    debugPrint('Page started loading: $url');
+  }
+
+  void _onPageFinished(String url) {
+    debugPrint('Page finished loading: $url');
+    final redirectedTo = Uri.parse(url);
+    final callbackUri = Uri.parse(widget.callbackUrl);
+    final bool hasReachedFinalRedirection =
+        redirectedTo.host == callbackUri.host;
+
+    if (hasReachedFinalRedirection) {
+      final queryParams = redirectedTo.queryParameters;
+      String? status = queryParams['status'];
+      String? message = queryParams['message'];
+      widget.on3dsDone(status, message);
+    }
+  }
+
+  void _onWebResourceError(WebResourceError error) {
+    debugPrint('''
+                Page resource error:
+                  code: ${error.errorCode}
+                  description: ${error.description}
+                  errorType: ${error.errorType}
+                  isForMainFrame: ${error.isForMainFrame}
+                  ''');
+  }
```

Comment:
Do we need `debugPrint` here? Especially for the error.
Perhaps we should pass the error to the application layer? So we can log it with `AseelLogger`?